cmd/server/pactasrv/conv/pacta_to_oapi.go

```diff
@@ -190,25 +200,31 @@ func IncompleteUploadToOAPI(iu *pacta.IncompleteUpload) (*api.IncompleteUpload,
 	if iu == nil {
 		return nil, oapierr.Internal("incompleteUploadToOAPI: can't convert nil pointer")
 	}
-	hd, err := HoldingsDateToOAPI(iu.HoldingsDate)
-	if err != nil {
-		return nil, oapierr.Internal("incompleteUploadToOAPI: holdingsDateToOAPI failed", zap.Error(err))
-	}
 	fc, err := FailureCodeToOAPI(iu.FailureCode)
 	if err != nil {
 		return nil, oapierr.Internal("incompleteUploadToOAPI: failureCodeToOAPI failed", zap.Error(err))
 	}
+	var hd *api.HoldingsDate
+	if iu.Properties.HoldingsDate != nil {
+		hd, err = HoldingsDateToOAPI(iu.Properties.HoldingsDate)
+		if err != nil {
+			return nil, oapierr.Internal("incompleteUploadToOAPI: holdingsDateToOAPI failed", zap.Error(err))
+		}
+	}
 	return &api.IncompleteUpload{
-		Id:                string(iu.ID),
-		Name:              iu.Name,
-		Description:       iu.Description,
-		HoldingsDate:      hd,
-		CreatedAt:         iu.CreatedAt,
-		RanAt:             timeToNilable(iu.RanAt),
-		CompletedAt:       timeToNilable(iu.CompletedAt),
-		FailureCode:       fc,
-		FailureMessage:    stringToNilable(iu.FailureMessage),
-		AdminDebugEnabled: iu.AdminDebugEnabled,
+		Id:                         string(iu.ID),
+		Name:                       iu.Name,
+		Description:                iu.Description,
+		CreatedAt:                  iu.CreatedAt,
+		RanAt:                      timeToNilable(iu.RanAt),
+		CompletedAt:                timeToNilable(iu.CompletedAt),
+		FailureCode:                fc,
+		FailureMessage:             stringToNilable(iu.FailureMessage),
+		AdminDebugEnabled:          iu.AdminDebugEnabled,
+		PropertyHoldingsDate:       hd,
+		PropertyESG:                optionalBoolToOAPI(iu.Properties.ESG),
+		PropertyExternal:           optionalBoolToOAPI(iu.Properties.External),
+		PropertyEngagementStrategy: optionalBoolToOAPI(iu.Properties.EngagementStrategy),
```

Comment:
It seems like an implementation detail that we internally store these things as an unstructured 'properties' blob, I'd drop the `Property` prefix on all of these. A HoldingsDate is a HoldingsDate regardless of how we store it in the DB.

And same throughout the API, both user-facing and at the `pacta` package level, I wouldn't lump all these properties under a `Properties` field, they have no more or less in common with each other than any other field (e.g. Name, Description) shared between multiple types.

For convenience, you could put all the shared fields between `IncompleteUpload` and `Porfolio` in some base struct and embed that.

Reply:
The reason I wanted to include this prefix is because I think it helps provide some helpful context which I think is useful to future engineers and API users. Properties differ from other fields of objects in these three ways:

1 - Our backend's role is exclusively to store them - we don't set these values, we don't base any of our business logic off of them, etc.
2 - Different Initiatives will have policies that require them to be set - a constraint that other data objects don't have.
3 - There are expected to be a pretty large number of them, and they're expected to expand over time (Hodie estimated ~1 per initiative onboarded onto the platform).

Because of these, wrapping these values in a shared struct has clear benefits:
- Unifies storage and retrieval (1)
- Makes expressions of constraint like (2) more self contained
- Reduces change costs for (3)

What do you think about that explanation?